pkg/liqo-controller-manager/core/foreigncluster-controller/status.go

```diff
@@ -89,15 +95,32 @@ func (r *ForeignClusterReconciler) handleNetworkingModuleStatus(ctx context.Cont
 				connectionErrorReason, connectionErrorMessage)
 		}
 	}
+	return nil
+}
+
+func (r *ForeignClusterReconciler) handleGatewaysStatus(ctx context.Context,
+	fc *liqov1beta1.ForeignCluster, statusExceptions map[liqov1beta1.ConditionType]statusException) error {
+	clusterID := fc.Spec.ClusterID
+
+	gwServer, errServer := getters.GetGatewayServerByClusterID(ctx, r.Client, clusterID)
+	gwClient, errClient := getters.GetGatewayClientByClusterID(ctx, r.Client, clusterID)
+
+	if errors.IsNotFound(errServer) && errors.IsNotFound(errClient) {
+		klog.V(6).Infof("Both GatewayServer and GatewayClient resources not found for ForeignCluster %q", clusterID)
+		statusExceptions[liqov1beta1.NetworkGatewayClientStatusCondition] = statusException{
```

Comment:
Defaulting to GatewayClient can be misleading if I expect to have a GatewayServer. I would use a new generic status condition or a better solution if any